scripts/llext_link_helper.py

```diff
@@ -33,21 +35,53 @@ def main():
 	elf = ELFFile(open(args.file, 'rb'))
 
 	text_addr = int(args.text_addr, 0)
+	p = re.compile('(^lib|\.so$)')
+	module = p.sub('', args.file)
 
-	text_offset = elf.get_section_by_name('.text').header.sh_offset
-	rodata_offset = elf.get_section_by_name('.rodata').header.sh_offset
-	data_offset = elf.get_section_by_name('.data').header.sh_offset
+	if elf.num_segments() != 0:
+		# A shared object type image, it contains segments
+		sections = ['.text', '.rodata', '.data', '.bss']
+		alignment = [0x1000, 0x1000, 0x10, 0x1000]
+	else:
+		# A relocatable object, need to handle all sections separately
+		sections = ['.text',
+			    f'._log_const.static.log_const_{module}_',
+			    '.static_uuids', '.z_init_APPLICATION90_0_', '.module',
+			    '.mod_buildinfo', '.data', '.trace_ctx', '.bss']
+		alignment = [0x1000, 0x1000, 0x0, 0x0, 0x0, 0x0, 0x10, 0x0, 0x1000]
 
-	upper = rodata_offset - text_offset + text_addr + 0xfff
-	rodata_addr = upper - (upper % 0x1000)
+	last_increment = 0
 
-	upper = data_offset - rodata_offset + rodata_addr + 0xf
-	data_addr = upper - (upper % 0x10)
+	command = [args.command]
+
+	for i in range(len(sections)):
+		try:
+			offset = elf.get_section_by_name(sections[i]).header.sh_offset
+			size = elf.get_section_by_name(sections[i]).header.sh_size
+		except:
```

Comment:
No, just catch the exception you're expecting. Copy/paste its name from the stack trace, it should take seconds. Also, use `pylint`.

https://pylint.pycqa.org/en/latest/user_guide/messages/warning/broad-exception-caught.html